knative-operator/deploy/resources/knativekafka/kafkachannel-v0.17.1.yaml

```diff
@@ -452,7 +452,7 @@ spec:
       serviceAccountName: kafka-ch-controller
       containers:
       - name: controller
-        image: gcr.io/knative-releases/knative.dev/eventing-contrib/kafka/channel/cmd/channel_controller@sha256:5fe9375f42e6353582acb5b70ae4470aef78f55d482d7809714085fc2e5b9782
+        image: registry.svc.ci.openshift.org/openshift/knative-v0.17.1:knative-eventing-sources-kafka-channel-controller
```

Comment:
@markusthoemmes  until we have the CSV override we need those, since we run here from a patch.

Reply:
Wondering why we would ship the CSV overrides after this? I would've expected a PR before this to ship them.

Reply:
tldr; Image override requires the CSV to have the changes and we didn't want to touch the CSV until we actually enable the controller.

Detail:
--------
CSV override doesn't work as is for Knative Kafka components. For eventing and serving, what we do is we just collect environment vars and pass them to upstream CR. Upstream operator will modify the deployments.
For Kafka components though, we need to modify the deployments ourselves in S-O. 

Another change is, there are some naming clashes. `DISPATCHER_IMAGE` env var is used both in eventing and in knative kafka components. I needed to add "scoping" for that (we once discussed about that).

So, these result in big changes.
- We didn't want to include them in this PR
- We also didn't want to do these in advance because of the "don't do it until you need it" principle (I made up the name but that's the idea I got from all the discussions in #serverless-dev)

Reply:
Got it, wfm.

Reply:
It's worth mentioning though that the problem with this is that `master` is not releasable after this PR. Without being able to replace variables via the CSV we can't productize and hence not release.

Reply:
I agree on this, but there will be a follow up PR of the CSV override soon.

afterwards we are back to a release-able state.